### PR TITLE
Add VkImageResolve VUIDs

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-VkImageResolve-srcImage-00268
VUID-VkImageResolve-srcImage-00271
VUID-VkImageResolve-srcImage-00273 
VUID-VkImageResolve-dstImage-00276
VUID-VkImageResolve-dstImage-00278

I wasn't able to find support on my any local GPUs for a 1D multi-sampled image so `00271` doesn't have a test... I doubt that anyone will actually hit `00271` as not sure who supports multi-sampled 1D images to do the resolve command from

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Running through CI.

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -3145,39 +3145,87 @@ bool CoreChecks::PreCallValidateCmdResolveImage(VkCommandBuffer commandBuffer, V
         // For each region, the number of layers in the image subresource should not be zero
         // For each region, src and dest image aspect must be color only
         for (uint32_t i = 0; i < regionCount; i++) {
-            skip |=
-                ValidateImageSubresourceLayers(cb_node, &pRegions[i].srcSubresource, "vkCmdResolveImage()", "srcSubresource", i);
-            skip |=
-                ValidateImageSubresourceLayers(cb_node, &pRegions[i].dstSubresource, "vkCmdResolveImage()", "dstSubresource", i);
-            skip |= VerifyImageLayout(cb_node, src_image_state, pRegions[i].srcSubresource, srcImageLayout,
+            const VkImageSubresourceLayers srcSubresource = pRegions[i].srcSubresource;
```
Comment (mark-lunarg) [inline layers/buffer_validation.cpp:3148]: Spencer, this passed CI and looks good, just had a nit and a suggestion.  First the nit: local vars should be lower_snake_case.

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -3145,39 +3145,87 @@ bool CoreChecks::PreCallValidateCmdResolveImage(VkCommandBuffer commandBuffer, V
...
+                    skip |= LogError(cb_node->commandBuffer, "VUID-VkImageResolve-srcImage-00268",
+                                     "vkCmdResolveImage(): pRegions[%u] baseArrayLayer must be 0 and layerCount must be 1 for all "
+                                     "subresources if the src or dst image is 3D.",
+                                     i);
+                }
+            }
+
+            if (VK_IMAGE_TYPE_1D == src_image_type) {
+                if ((pRegions[i].srcOffset.y != 0) || (pRegions[i].extent.height != 1)) {
+                    skip |= LogError(cb_node->commandBuffer, "VUID-VkImageResolve-srcImage-00271",
```
Comment (mark-lunarg) [inline layers/buffer_validation.cpp:3199]: Not so sure about the one above, but for the following messages would it not be useful to output the image also?  If you agree, please use a `LogObjectList ` container for the CB and image.

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:3199]: is `LogObjectList` new or am I just that blind?
Happy to add for all these new VUIDs as with many command buffers and images in flight, this is actually a good use for it IMO as well

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp:3199]: Also not sure if `report_data->FormatHandle(object).c_str()` is more useful or not, that is what I have normally tried using in the past

Comment (mark-lunarg) [reply layers/buffer_validation.cpp]: @sfricke-samsung, I'm sorry, I think I mislead you.  You actually need both -- the `LogObjectList `is to ensure the objects are returned as parameters in the `DebugUtils `callback.  The `FormatHandle `stuff is to have them printed in the local message itself -- the best practice is to do both.  The objectlist is _fairly_ new, but multi-object messages are not as common as you'd expect.  I again apologize!

Comment (sfricke-samsung) [reply layers/buffer_validation.cpp]: no problem, will add the `FormatHandle` in the message as well

**Comment by mark-lunarg** [comment]
Looks good!  Running through CI.

**Pull request merged**